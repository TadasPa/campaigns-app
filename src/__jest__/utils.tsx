import React, { PropsWithChildren, ReactElement } from "react";
import { render } from "@testing-library/react";
import { ThemeProvider, DefaultTheme } from "styled-components";
import { theme } from "../theme";
import GlobalStyles from "../theme/globalStyles";

const Providers = (initialTheme: DefaultTheme) => ({
  children,
}: PropsWithChildren<any>) => (
  <ThemeProvider theme={initialTheme}>
    <GlobalStyles />
    {children}
  </ThemeProvider>
);

interface ICustomRenderOptions {
  initialTheme?: DefaultTheme;
}

const customRender = (
  ui: ReactElement,
  { initialTheme = theme as any, ...restOptions }: ICustomRenderOptions = {}
) =>
  render(ui, {
    wrapper: Providers(initialTheme),
    ...restOptions,
  });

export * from "@testing-library/react";

export { customRender as render };
