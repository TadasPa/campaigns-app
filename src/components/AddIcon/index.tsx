import React, { FC, HTMLAttributes } from "react";
import { ReactComponent as AddSVG } from "../../assets/icons/plus-circle-fill.svg";
import styled from "styled-components";
import { FlexBox } from "../Grid";

const AddIcon: FC<HTMLAttributes<HTMLDivElement>> = ({ children, ...rest }) => {
  return (
    <FlexBox alignItems="center" {...rest}>
      <AddSVG title="Add Items" />
      {children}
    </FlexBox>
  );
};

const StyledAddIcon = styled(AddIcon)`
  & svg {
    width: 30px;
    height: 30px;
    margin-right: 5px;
    fill: ${({ theme: { colors } }) => colors.buttons.primary};
  }

  &:hover {
    cursor: pointer;
    opacity: 0.8;
  }
`;

export default StyledAddIcon;
