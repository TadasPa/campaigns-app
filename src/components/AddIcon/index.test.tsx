import React from "react";
import AddIcon from ".";
import { render } from "../../__jest__/utils";

test("AddIcon machtes snapshot", () => {
  const { container } = render(<AddIcon />);
  expect(container).toMatchSnapshot();
});
