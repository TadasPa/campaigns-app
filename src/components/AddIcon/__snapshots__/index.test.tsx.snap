// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddIcon machtes snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 svg {
  width: 30px;
  height: 30px;
  margin-right: 5px;
  fill: #3149b9;
}

.c1:hover {
  cursor: pointer;
  opacity: 0.8;
}

<div>
  <div
    class="c0 c1"
  >
    <svg
      title="Add Items"
    >
      plus-circle-fill.svg
    </svg>
  </div>
</div>
`;
